### PR TITLE
Use secure CSPRNG

### LINKED ISSUE #1175: Use a CSPRNG to generate judge keys
On https://dmoj.ca/admin/judge/judge/add/, the "regenerate" button show in the picture uses `Math.random` instead of a CSPRNG:

This disturbs me.

### PR DESCRIPTION
**Pull request by ehhthing**
Fixes #1175

### CODE REVIEW

File: judge/admin/runtime.py [hunk 1]
```diff
@@ -57,9 +57,13 @@ def render(self, name, value, attrs=None, renderer=None):
...
             charset = "abcdefghijklnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789`~!@#$%^&*()_+-=|[]{{}};:,<>./?",
-            key = "";
+            key = "",
+            random = new Uint8Array(length);
+        
+        window.crypto.getRandomValues(random);
+
         for (var i = 0, n = charset.length; i < length; ++i) {{
-            key += charset.charAt(Math.floor(Math.random() * n));
+            key += charset.charAt(random[i] % n);
```
Comment (quantum5) [inline judge/admin/runtime.py]: This distribution is just wrong. The first 71 characters of the charset are 50% more likely to appear than the last 21 characters.

**Comment by TheAvidDev** [comment]
Shouldn't something more standard like python's [secrets](https://docs.python.org/3/library/secrets.html) library be used?

**Comment by ehhthing** [comment]
> Shouldn't something more standard like python's [secrets](https://docs.python.org/3/library/secrets.html) library be used?

I'm not exactly sure what you are suggesting here. This is fully client side, meaning that we can only use the JavaScript [Web Crypto API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API) which is a standard.

**Comment by quantum5** [comment]
Appears to have been superseded by #1757.

**Pull request closed without merging**